lib/browser/api/browser-window.js

```diff
@@ -103,6 +104,15 @@ BrowserWindow.prototype._init = function () {
     app.emit('browser-window-focus', event, this)
   })
 
+  // Exit HTML fullscreen when window leaves fullscreen
+  this.on('leave-full-screen', (event) => {
+    for (const webContents of getAllWebContents()) {
+      if (webContents.getOwnerBrowserWindow() === this) {
+        webContents.executeJavaScript('document.webkitExitFullscreen()')
```

Comment:
We can check `is_html_fullscreen()` before executing the script, the API needs to be exposed in JS though.